bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/handlers/EntityBaseHandler.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.handlers;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.paradoxalarm.internal.exceptions.ParadoxBindingException;
+import org.openhab.binding.paradoxalarm.internal.model.ParadoxPanel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EntityBaseHandler} abstract handler class that contains common logic for entities.
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public abstract class EntityBaseHandler extends BaseThingHandler {
+
+    private static final long INITIAL_DELAY_SECONDS = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(EntityBaseHandler.class);
+
+    protected EntityConfiguration config;
+
+    public EntityBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing.");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        config = getConfigAs(EntityConfiguration.class);
+
+        scheduler.schedule(this::initializeDelayed, INITIAL_DELAY_SECONDS, TimeUnit.SECONDS);
+    }
+
+    private void initializeDelayed() {
+        logger.trace("Start initializeDelayed() in {}", getThing().getUID());
+        try {
+            ParadoxPanel panel = ParadoxPanel.getInstance();
+            if (!panel.isPanelSupported()) {

```

Comment:
Where do host not founds and other communication errors end up?

Reply:
Everything ends up in the catch clause of initializeCommunication() method. If something goes wrong the bridge status gets updated to offline with the error from the catched exception.